### PR TITLE
Fix role double-join errors

### LINKED ISSUE #49: Joing role you're already a member of doesn't give proper response
**Description**
If a user tries to join a role they're already a member of, the bot replies that the user has now joined the role rather than that the user is already in the role. This is confusing and doesn't reflect what the bot is doing in that case (no-op).

**To Reproduce**
```
!role create poo
!role join poo    <-- bot replies "user has joined poo", which is correct
!role join poo    <-- bot replies "user has joined poo", should reply "user is already a member of poo"
```

**Expected behavior**
Better response to indicate the no-op

**Screenshots**
no

**Additional context**
no no

### PR DESCRIPTION
**Pull request by nhawke**
Now, when a user joins a role they are already a member of, a proper
error message is sent instead of a duplicate success message.

fixes #49

### CODE REVIEW

**Review by RobertWLeslie — APPROVED** [review]
lg2m

File: src/commands/role/__init__.py [hunk 1]
```diff
@@ -63,8 +63,8 @@ def permission_failure_message(action: str, target_name: str) -> CommandOutput:
     :return: A CommandOutput with the created message.
     """
     # TODO: factor this out to be common to all commands
-    return CommandOutput(command_status=status.FAIL).set_text(f"Memebot doesn't have permission to {action} role {target_name}. "
-                                                       "Are you sure you configured Membot's permissions correctly?")
+    return CommandOutput(command_status=status.FAIL).set_text(f"Memebot doesn't have permission to {action} role `@{target_name}`. "
```
Comment (super-cooper) [inline src/commands/role/__init__.py:66]:
```python3
    return CommandOutput(command_status=status.FAIL,
                         content=f"Memebot doesn't have permission to {action} role `@{target_name}`. "
                                 "Are you sure you configured Membot's permissions correctly?")
```

**Review by super-cooper — CHANGES_REQUESTED** [review]
Looks good just trying to squeeze that one line under 120 characters

File: src/commands/role/__init__.py (hunk 1 above)
Comment (nhawke) [reply src/commands/role/__init__.py]: consider it done

**Review by super-cooper — APPROVED** [review]
lgtm

**Review by RobertWLeslie — APPROVED** [review]
lg2m

**Pull request merged**